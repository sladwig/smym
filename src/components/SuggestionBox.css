.suggestion-box {
    display: flex;
    flex-direction: column;
    flex: 0 0 auto;
    width: 100%;
}

.suggestion {
    padding-left: 111px;
    height: 90px;
    display: flex;
    align-items: center;
    transition: background-color 125ms ease-in-out;
}

.suggestion.active {
    background-color: var(--pale-grey);
}

.place-suggestion {
    color: white;
    border-radius: 32px;
    font-size: 24px;
    padding-top: 14px;
    padding-bottom: 14px;
    padding-left: 19px;
    padding-right: 19px;
    animation: suggestion-appear 0.25s cubic-bezier(0.230, 1.000, 0.320, 1.000) ;
}

.suggestion-info {
    border-top: 1px solid var(--pale-grey);
    animation: suggestion-appear 0.45s cubic-bezier(0.230, 1.000, 0.320, 1.000) ;
}


@keyframes suggestion-appear {
  0% {
    transform: scale(0.3) translateY(-40px);
    opacity: 0;
  }
  100% {
    transform: scale(1) translateY(0px);
    opacity: 1;
  }
}
