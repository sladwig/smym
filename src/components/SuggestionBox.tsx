import React from 'react';
import { create } from 'zustand';
import { store } from '../store/Store';
import fuzzysearch from 'fuzzysearch';
import './SuggestionBox.css';
import { IPlace } from '../store/Place';

interface IProps {}

export const SuggestionBox = ({}: IProps) => {
    const mode = useSuggestionStore(state => state.mode);
    const value = useSuggestionStore(state => state.value);

    return suggestionFor(mode, value);
};

type SuggestionMode = 'none' | 'places';

const suggestionFor = (mode: SuggestionMode, searchTerm: string) => {
    if (mode === 'none') return null;
    if (mode !== 'places') return null;

    const filteredSuggestions = [...store.placesList].filter(p => fuzzysearch(searchTerm, p.name));

    suggestionStore.setState({ length: filteredSuggestions.length });

    return <div className="suggestion-box">{filteredSuggestions.map(Suggestion)}</div>;
};

const Suggestion = ({ name, color: backgroundColor }: IPlace) => (
    <div className="suggestion">
        <div className="place-suggestion" style={{ backgroundColor }}>
            {name}
        </div>
    </div>
);

const [useSuggestionStore, suggestionStore] = create(set => ({
    mode: 'none',
    value: '',
    length: 0,
    reset: () => set({ mode: 'none', value: '' }),
}));

export { useSuggestionStore, suggestionStore };
