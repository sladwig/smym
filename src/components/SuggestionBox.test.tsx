import { render } from '@testing-library/react';
import React from 'react';
import { SuggestionBox } from './SuggestionBox';

describe('SuggestionBox', () => {
    test('renders', () => {
        const { container } = render(<SuggestionBox />);
        expect(container).toMatchSnapshot();
    });
});
