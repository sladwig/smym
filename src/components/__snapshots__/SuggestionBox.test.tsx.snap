// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionBox renders 1`] = `
<div>
  <div
    class="suggestion-box"
  >
    <div>
      a suggestion
    </div>
  </div>
</div>
`;
